docs/operator.md

```diff
@@ -2049,6 +2049,82 @@ Hostnames for [Kubernetes host aliases  :octicons-link-external-16:](https://kub
+
+### `roles.role`
+
+The [cusom MongoDB role :octicons-link-external-16:](https://www.mongodb.com/docs/manual/core/security-user-defined-roles/) name.
+
+| Value type | Example |
+| ---------- | ------- |
+| :material-code-string: string | `myClusterwideAdmin` |
+
+### `roles.db`
+
+Database in which you want to store the user-defined role.
+
+| Value type | Example |
+| ---------- | ------- |
+| :material-code-string: string | `admin
+
+### `roles.authenticationRestrictions.clientSource`
+
+List of the IP addresses or CIDR blocks *from which* users assigned this role can connect. MongoDB servers reject connection requests from users with this role if the requests come from a client that is not present in this array.
+
+| Value type  | Example    |
+| ----------- | ---------- |
+| :material-text-long: subdoc      | `127.0.0.1` |
+
+### `roles.authenticationRestrictions.serverAddress`
+
+List of the  IP addresses or CIDR blocks *to which* users assigned this role can connect. MongoDB servers reject connection requests from users with this role if the client requests to connect to a server that is not present in this array.
+
+| Value type  | Example    |
+| ----------- | ---------- |
+| :material-text-long: subdoc      | `127.0.0.1` |
+
+### `roles.privileges.actions`
+
+List of custom role actions  that users granted this role can perform: For a list of accepted values, see [Privilege Actions :octicons-link-external-16:](https://www.mongodb.com/docs/manual/reference/privilege-actions/#database-management-actions) in the MongoDB Manual.
+
+| Value type  | Example    |
+| ----------- | ---------- |
+| :material-text-long: subdoc      | `addShard` |
+
+### `roles.privileges.resource.db`
+
+Database for which the custom role actions apply. An empty string ("") indicates that the privilege actions apply to all databases.
+
+| Value type | Example |
+| ---------- | ------- |
+| :material-code-string: string | `""` |
+
+### `roles.privileges.resource.collection`
+
+Collection for which the custom role actions apply. An empty string ("") indicates that the privilege actions apply to all of the database's collections.
+
+| Value type | Example |
+| ---------- | ------- |
+| :material-code-string: string | `""` |
+
+### `roles.privileges.resource.cluster`
+
+If true, the custom role actions apply to all databases and collections in the MongoDB deployment. False by default. If set to true, values for `roles.privileges.resource.db` and `roles.privileges.resource.collection` shouldn't be provided.
+
+| Value type  | Example    |
+| ----------- | ---------- |
+| :material-toggle-switch-outline: boolean     | `true`     |
+
+### `roles.roles`
+
+An array of roles (with names of the role and the database) from which this role inherits privileges. The field is mandatory. Empty array (`[]`) can be used to specify no roles to inherit from.
```

Comment:
This is not mandatory in our implementation. (Spec was updated recently.)

Reply:
fixed